ui_testing/testcases/basic_tests/test006_orders.py

```diff
@@ -3657,4 +3657,55 @@ def test104_edit_order_page_then_overview(self, edit_case):
         else:
             self.assertFalse(self.order_page.confirm_popup(check_only=True))
             self.info('asserting redirection to active table')
-            self.assertEqual(self.order_page.orders_url, self.base_selenium.get_url())
\ No newline at end of file
+            self.assertEqual(self.order_page.orders_url, self.base_selenium.get_url())
+
+
+    def test107_check_that_two_testunits_with_same_name_displayed_in_one_testplan(self) :
+            """
+           Orders: Test plan Approach: test units pop-up: In case I have two test units with the same name
+           in one test plan, both of them should display in the test units pop-up.
+            LIMS-4800
+            """
+            self.test_plan_api = TestPlanAPI()
+            self.analysis_page =SingleAnalysisPage()
+            tu_name = self.generate_random_string()
+            response, payload = self.test_unit_api.create_qualitative_testunit(name=tu_name)
+            first_tu = self.test_unit_api.get_testunit_form_data(response['testUnit']['testUnitId'])[0]['testUnit']
+            formatted_tu = TstUnit().map_testunit_to_testplan_format(testunit=first_tu)
+            response2, payload2 = self.test_unit_api.create_qualitative_testunit(name=tu_name)
+            second_tu = self.test_unit_api.get_testunit_form_data(response2['testUnit']['testUnitId'])[0]['testUnit']
+            formatted_tu_2 = TstUnit().map_testunit_to_testplan_format(testunit=second_tu)
+            tp_response, tp_payload = self.test_plan_api.create_testplan(testUnits=[formatted_tu,formatted_tu_2],selectedTestUnits=[formatted_tu,formatted_tu_2])
+            # testplan={'id': tp_payload['testPlan']['id'],'name':tp_payload['testPlan']['text']}
```

Comment:
testPlan need to be :
[{'testPlan': {'id': testplan['id'], 'text': testplan['name']}}]